### PR TITLE
Exploratory Analysis 

### PR DESCRIPTION
**Pull request by MikeLippincott**
Here I cluster data via umap for plate 2 and show that there is no cell count drop off across treatments and dosages. 

If reviewer(s) could please look past the tsne stuff for plate1 (wave 1-3) and focus on plate 2 analysis and cell count analysis that would be phenomenal. Thank you in advance!

### CODE REVIEW

File: 1.Exploratory_Data_Analysis/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# This folder hosts jupyter notebooks for exploratory data analysis (EDA) for multiple 'waves' of data. See this repo's main page for a link to the Interstellar preprocessing page.
```
Comment (axiomcura) [inline 1.Exploratory_Data_Analysis/README.md:1]: This may be defined else where. If not, considered providing a definition about what `waves` means for contextual purposes.

File: 1.Exploratory_Data_Analysis/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+```Exploratory_Data_Analysis.ipynb```
+is for clustering of cells across all features. The goal is to aim to see if treatments cluster or not.
```
Comment (axiomcura) [inline 1.Exploratory_Data_Analysis/README.md:4]:
```suggestion
is for clustering cells across all features. The goal is to aim and see if the treatments cluster or not.
```

Feel free to reject suggestions.

File: 1.Exploratory_Data_Analysis/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+```cell_count_analysis.ipynb``` is for qc analysis to assess if there are cell count drop offs for a given treatment.
```
Comment (axiomcura) [inline 1.Exploratory_Data_Analysis/README.md:5]: Is `qc` defined else where? If not, consider providing the full name first then use abbreviations.

File: 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_Disulfiram.png
Comment (axiomcura) [inline 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_Disulfiram.png:1]: Consider adding units into your graphs

File: 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_Flagellin.png
Comment (axiomcura) [inline 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_Flagellin.png:1]: same comment above

File: 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_H2O2.png
Comment (axiomcura) [inline 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_H2O2.png:1]: same comment above

File: 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_LPS.png
Comment (axiomcura) [inline 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_LPS.png:1]: same comment above

File: 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_Thapsigargin.png
Comment (axiomcura) [inline 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_Thapsigargin.png:1]: same comment above

File: 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_media ctr.png
Comment (axiomcura) [inline 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_media ctr.png:1]: What does `0` in the x-axis mean? Is this a place holder for a name soon to come?

File: 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_treatment.png
Comment (axiomcura) [inline 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_treatment.png:1]: Are you counting the number of cells across all concentrations or a specific concentration? If it is cells from a specific concentration, consider adding the amount.

File: 1.Exploratory_Data_Analysis/nbconverted/Exploratory_Data_Analysis.py [hunk 4]
```diff
@@ -20,26 +20,29 @@
...
 
 get_ipython().run_line_magic("matplotlib", "inline")
+import itertools
 
 from sklearn.cluster import KMeans
 from sklearn.manifold import TSNE
 
 sys.path.append("..")
-from ..utils.utils import *
+from utils.utils import df_stats, plot_features_all_cluster
```
Comment (axiomcura) [inline 1.Exploratory_Data_Analysis/nbconverted/Exploratory_Data_Analysis.py:39]: Make sure you follow import order: 
1 - standard lib
2 - 3rd party lib 
3 - file based imports (user made imports)
[source](https://peps.python.org/pep-0008/#imports)

File: 1.Exploratory_Data_Analysis/nbconverted/cell_count_analysis.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+get_ipython().run_line_magic("matplotlib", "inline")
+import itertools
+import pathlib
+
+import pyarrow.parquet as pq
+from sklearn.cluster import KMeans
+from sklearn.manifold import TSNE
+
+sys.path.append("..")
+from utils.utils import df_stats, plot_features_all_cluster
```
Comment (axiomcura) [inline 1.Exploratory_Data_Analysis/nbconverted/cell_count_analysis.py:23]: same comment as above

File: 1.Exploratory_Data_Analysis/nbconverted/cell_count_analysis.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    plt.xlabel(i)
+    plt.ylabel("Number of single cells")
+    plt.xticks(rotation=45)
+    plt.title(f"Number of single cells per {i}")
+    plt.savefig(
+        f"Figures/cell_counts_plate2/Number_of_single_cells_per_{i}.png",
+        bbox_inches="tight",
+    )
+    plt.show()
+    plt.close()
```
Comment (axiomcura) [inline 1.Exploratory_Data_Analysis/nbconverted/cell_count_analysis.py:80]: considered providing some comment explaining what data is being selected in order to generate these plots.

**Review by axiomcura — APPROVED** [review]
Nice PR. 

Left some commens and suggestions. Let me know if you have any questions!

File: 1.Exploratory_Data_Analysis/nbconverted/cell_count_analysis.py (hunk 6 above)
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/nbconverted/cell_count_analysis.py:80]: added

File: 1.Exploratory_Data_Analysis/nbconverted/cell_count_analysis.py (hunk 5 above)
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/nbconverted/cell_count_analysis.py:23]: fixed via isort

File: 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_Disulfiram.png
Comment (MikeLippincott) [reply 1.Exploratory_Data_Analysis/Figures/cell_counts_plate2/Number_of_single_cells_per_Disulfiram.png:1]: this might be another PR
